app/models/manageiq/providers/ibm_cloud/inventory/parser/power_virtual_servers.rb

```diff
@@ -307,12 +307,17 @@ def availability_zones
 
   def snapshots
     collector.snapshots.each do |snapshot|
+      sum = 0 # API not returning size of snapshot, thus have to tally up size of associated volumes
+      snapshot.volume_snapshots.each_key do |vol|
+        sum += collector.volume(vol).size&.gigabytes
+      end
```

Comment:
Slightly simpler for readability:

```suggestion
      # API not returning size of snapshot, thus have to tally up size of associated volumes
      snapshot.volume_snapshots.keys.map { |vol| collector.volume(vol).size&.gigabytes.to_i }.sum
```

Reply:
I think you can use the `sum` directly as the enumerator method: `total_size = snapshot.volume_snapshots.keys.sum { |vol| collector.volume(vol).size&.gigabytes.to_i }`

Reply:
Oh yes you are right...this is better...I'll update my suggestion

Reply:
Love ruby, I just thought "wouldn't it be cool if you could just .sum the result of the block....let me try that and see if it works"

Reply:
Sure, looks simpler, but I get;
```
ManageIQ::Providers::BaseManager::Refresher::PartialRefreshError (undefined local variable or method `sum' for #<ManageIQ::Providers::IbmCloud::Inventory::Parser::PowerVirtualServers::CloudManager:0x0000000023d32cc8>)
```
Could you help this Ruby newbie? 🙂

Reply:
@Fryguy's example sets the result to `total_size` not `sum` so you'd have to change below from `:total_size     => sum,` to `:total_size     => total_size`

Reply:
I was simply confused by the `sum` reference above being the method or the local variable. Revised code force-pushed. Please review again.

Reply:
Ah yeah sorry about that...forgot I changed the variable name to something less generic.

Reply:
Latest force-push should have everything discussed above. Please take a look!